### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by des-des**
issue #147 

Okay so here is basic integration with auberauth + gh, we still have not configured an auth plug to make sure  to attach to routes we want logged in users for, but we can add that with the routes for the issues views.

@nelsonic, I was getting runtime errors when testing https://github.com/dwyl/dwylbot/blob/master/web/models/user_from_auth.ex#L25 / https://github.com/dwyl/tudo/pull/163/commits/2a559356474e5e8d4414688357abf001f1dd3866#diff-1be47e13d25b7a9070d11578fa5ce415R25.

It seems this method for accessing keys throws when the do not exist.. Thoughts?

### CODE REVIEW

**Comment by nelsonic** [comment]
@Jbarget please ***assign*** to me when you're happy.
@des-des sorry I did not _see_ this PR.
My "Todo List" (_for reviewing PRs_) is: https://github.com/pulls?utf8=%E2%9C%93&q=is%3Aopen+is%3Apr+assignee%3Anelsonic
so the "request review" GitHub feature does not put the PR on my "radar"... 😞

**Comment by des-des** [comment]
@nelsonic yh sorry my bad, I also only look at assigned reviews, not sure why I didnt assign (done now)

**Review by nelsonic — APPROVED** [review]
@des-des this code LGTM. 👍 

(_I would prefer to have Auth in a re-useable sub-module but for now this will get us up-and-running._)

The _reason_ I have not reviewed/merged this PR is because it's assigned to two people. 
I _thought_ it was being "_peer-reviewed_" by @Jbarget `before` being assigned to me for QA/Merge. that's the workflow I _thought_ we were using for this project...? 🤔 
Perhaps it's worth having the workflow clarified?
From my side this PR is good to merge
so I'm un-assigning myself (_for now_) until @Jbarget leaves his review. (_please feel free to assign back to me for merging or merge if you're happy_) thanks.

**Comment by Jbarget** [comment]
@nelsonic sure thing will review at next available opportunity

NB. I wasnt aware of review workflow

**Comment by nelsonic** [comment]
@Jbarget cool. (_know you're busy so any time you can!_)
We appear to not have a `CONTRIBUTING.md` in this repo yet ...
https://github.com/dwyl/tudo/tree/929281957f4770f3714d66912588ea15b07eaacd ...
Also, the "_peer review_" workflow is not well defined.
CC: @iteles where should I open the issue to address this?

**Comment by iteles** [comment]
@nelsonic On the contributing repo for the peer review workflow and on this repo for adding a CONTRIBUTING.md file!

File: web/models/user_from_auth.ex [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    %{id: id, name: name_from_info(info), avatar: avatar}
+  end
+
+  defp name_from_info(info) do
+    case info do
+      %{name: name} when not is_nil(name) and name != "" ->
+        name
+    _ ->
+      name = [info.first_name, info.last_name]
```
Comment (Jbarget) [inline web/models/user_from_auth.ex:16]: in what situation is there a first_name and last_name from github?

Comment (nelsonic) [reply web/models/user_from_auth.ex:16]: Most people who take GH seriously have a first and last name on their profile.
In the GH API however this is just `"name"` so we may need to _split_ on `' '` or simply store it as `"name"` ...

**Comment by nelsonic** [comment]
Nice one gents! 👍 
Hope you have a great weekend! 🎉

**Pull request merged**